Simplified/AxisService/NYPLRSACypher.swift

```diff
@@ -126,52 +128,131 @@ struct NYPLRSACypher: NYPLRSACryptographing {
     
     self.publicKey = publicKey.base64EncodedString()
     self.privateKey = privateKey.base64EncodedString()
-    self.modulus = self.publicKey.replacingOccurrences(of: "/", with: "-")
-    // we're creating a key with 2048 bits. Exponent for that is AQAB.
-    self.exponent = "AQAB"
-  }
-  
-  func encryptText(_ message: String) -> String? {
-    let _messageData = message.data(using: .utf8)
     
-    guard let messageData = _messageData else {
+    guard
+      let (mod, exp) = NYPLRSACypher.parsePublicSecKey(publicKey: self.publicKeySec)
+    else {
       NYPLRSACypher
-        .logFailedEncryptionError(NYPLRSACypher.stringToDataConversionFailure)
+        .logFailedInitializationError(
+          "Failed to get exponent and or modulus for public key")
       return nil
     }
-    guard let encryptedData = SecKeyCreateEncryptedData(publicKeySec,
-                                                        .rsaEncryptionOAEPSHA1,
-                                                        messageData as CFData,
-                                                        nil)
-      else {
-        NYPLRSACypher
-          .logFailedEncryptionError(NYPLRSACypher.nilEncryptedDataFailure)
-        return nil
+    
+    /// The minimum length of base64 string generated from mod data should be 342. Anything less and the
+    /// license file we get will be unusable since we won't be able to decrypt bookVaultId and AES key in our
+    /// license file using our private key.
+    let modulus = mod.base64EncodedString()
+    guard modulus.count >= 342 else {
+      NYPLRSACypher
+        .logFailedInitializationError("Got incorrect modulus from public key")
+      return nil
     }
     
-    return (encryptedData as Data).base64EncodedString()
+    self.modulus = String(modulus.prefix(342)).replacingOccurrences(of: "/", with: "_")
+    self.exponent = exp.base64EncodedString()
   }
   
-  func decryptText(_ encryptedString: String) -> String? {
-    let _messageData = Data(base64Encoded: encryptedString)
+  /**
+   `This is how this method works:`
+   
+   We first extract a CFDictionary from the public key, from that key we get the key size and key data. Key data
+   content is DER-encoded ASN for
+   
+   SEQUENCE {
+    modulus INTEGER,
+    publicExponent INTEGER
+   }
+   
+   These integers are padded with a zero byte if they are positive, and have a leading 1-bit. Therefore we
+   might get one extra byte. If that happens, we just cut it away.
+   
+   Here's what the data looks like
+   First 8 bits - some bits we do not need. Anything after that upto the 5th last bit is modulus. And the last 3
+   bits are the exponent.
+   
+   Our public key is 2048 bits long. Modulus for a public key is 1/8th the key size. But since we might have a
+   leading 1-bit, in which case our modulus will be 257 bits, we remove the extra leading bit.
```

Comment:
from reading the SO answer, is it possible you meant "257 _bytes_", and then "we remove the extra leading _byte_"?